### PR TITLE
Ability to configure Docker daemon options

### LINKED ISSUE #90: Feature Request: allow override of dockerd options
I love your Ansible Roles -- but the major no-go for me is the fact I can't override the storage driver easily.

### PR DESCRIPTION
**Pull request by Rickkwa**
Resolves #90

### CODE REVIEW

**Comment by davidecavestro** [comment]
This patch would also cover #84

**Comment by varac** [comment]
Hej, what's the state of this PR ? Would love to see it merged soon !

**Comment by llbbl** [comment]
devicemapper shouldn't be set as default.

https://github.com/docker/cli/pull/1424

**Comment by boxrick** [comment]
This would also fix some issues I am having, @geerlingguy any chance of a merge?

**Comment by andrey1vanov** [comment]
@geerlingguy is there any chance that PR will be merged? As this is a really nice feature.

**Comment by robbyoconnor** [comment]
Please reopen this

**Comment by robbyoconnor** [comment]
Please reopen this

**Comment by dr-edmond-merkle** [comment]
I think you need to create the configuration file before the installation of docker in the task? I needed that for this to work for me locally.

File: tasks/main.yml [hunk 1]
```diff
@@ -17,6 +17,18 @@
     state: "{{ docker_service_state }}"
     enabled: "{{ docker_service_enabled }}"
 
+- name: Ensure /etc/docker/ directory exists.
+  file:
+    path: /etc/docker
+    state: directory
```
Comment (geerlingguy) [inline tasks/main.yml:23]: The last thing I'd like to make sure to do is _not_ touch the file if the default (`docker_daemon_options: {}`) is set.

That way (a) existing role users won't have that file wiped out if it currently contains things and they update the role, and (b) people can manage the file on their own simply by _not_ setting the option in this role's vars.

**Review by geerlingguy — CHANGES_REQUESTED** [review]

**Comment by robbyoconnor** [comment]
bump

**Comment by robbyoconnor** [comment]
bump

**Comment by geerlingguy** [comment]
Closing in favor of #256, which is based on this.

**Pull request closed without merging**